File: src/mcp/chunk.ts
export function chunkText(
  fullText: string,
  maxChars: number = 500,
  overlap: number = 50
): string[] {
  const chunks: string[] = [];
  let start = 0;

  while (start < fullText.length) {
    const end = Math.min(start + maxChars, fullText.length);
    const chunk = fullText.slice(start, end).trim();

    if (chunk.length > 0) {
      chunks.push(chunk);
    }

    start += maxChars - overlap;
  }

  return chunks;
}

File: src/mcp/chunk.py
def chunk_text(full_text: str, max_chars: int = 500, overlap: int = 25) -> list[str]:
    """Chunk text into overlapping pieces."""
    chunks: list[str] = []
    start = 0

    while start < len(full_text):
        end = min(start + max_chars, len(full_text))
        chunk = full_text[start:end].strip()

        if len(chunk) > 0:
            chunks.append(chunk)

        start += max_chars - overlap

    return chunks
